File: code/mathematical-algorithms/factorial/factorial.rs
use std::env;

// Part of Cosmos by OpenGenus Foundation

fn factorial(n: i64) -> i64 {
    if n == 0 {
        return 1;
    } else {
        return n * factorial(n-1);
    }
}

fn main() {
    if let Some(arg) = env::args().nth(1) {
        if let Ok(x) = arg.parse::<i64>() {
            println!("{}", factorial(x));
        }
    }
}

File: code/mathematical-algorithms/factorial/factorial.kt
// Part of Cosmos by OpenGenus Foundation

fun factorial(n: Int): Int {
    return if (n == 0) {
        1
    } else {
        n * factorial(n-1)
    }
}

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val result = factorial(n)
    println(result)
